pkg/ingester/ingester.go

```diff
@@ -354,6 +358,43 @@ func (i *Ingester) QuerySample(req *logproto.SampleQueryRequest, queryServer log
 	return sendSampleBatches(queryServer.Context(), heapItr, queryServer)
 }
 
+func (i *Ingester) GetChunkIDs(ctx context.Context, req *logproto.GetChunkIDsRequest) (*logproto.GetChunkIDsResponse, error) {
+	orgID, err := user.ExtractOrgID(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	// when using boltdb-shipper, limit the request start time to the start of current schema config otherwise
+	// if query covers non boltdb-shipper stores too then we would un-necessarily be querying them as well which would anyways be done by queriers.
+	reqStart := req.Start
+	if i.store.ActivePeriodConfig().IndexType == shipper.BoltDBShipperType {
```

Comment:
Is there an edge case here if the previous schema was also boltdb-shipper? Should we instead just cap the query time to the maxlookback?

Reply:
Yes, there is an edge case but it should not be a problem after running boltdb shipper for a few hours to have enough data for current schema. We do not always set `maxlookback` anymore which otherwise was. I kept it this way to reduce the complexity but can change it to set `maxlookback` for this case as well while keeping `querystore` to false if you want.

Reply:
I think we should change this, if there were a schema change without a restart and the previous and new schema were both boltdb-shipper this would create a window where logs would be missing from queries as we would need to query the previous boltdb-shipper index for files which haven't been downloaded by queriers but wouldn't be able to because of this truncation.

I haven't really thought through what solution makes sense to me yet but am thinking we should just fix this at max chunk age plus some amount of time.